src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java

```diff
@@ -182,35 +193,13 @@ public void stop(@NonNull Throwable cause) throws Exception {
     }
 
     @Override public void onResume() {
-        super.onResume();
-        // See if we are still running, or scheduled to run. Cf. stop logic above.
         try {
-            Run<?, ?> run = getContext().get(Run.class);
-            for (Queue.Item item : Queue.getInstance().getItems()) {
-                if (item.task instanceof PlaceholderTask && ((PlaceholderTask) item.task).context.equals(getContext())) {
-                    LOGGER.log(FINE, "Queue item for node block in {0} is still waiting after reload", run);
-                    return;
-                }
-            }
-            Jenkins j = Jenkins.getInstanceOrNull();
-            if (j != null) {
-                for (Computer c : j.getComputers()) {
-                    for (Executor e : c.getExecutors()) {
-                        Queue.Executable exec = e.getCurrentExecutable();
-                        if (exec instanceof PlaceholderTask.PlaceholderExecutable && ((PlaceholderTask.PlaceholderExecutable) exec).getParent().context.equals(getContext())) {
-                            LOGGER.log(FINE, "Node block in {0} is running on {1} after reload", new Object[] {run, c.getName()});
-                            return;
-                        }
-                    }
-                }
-            }
-            TaskListener listener = getContext().get(TaskListener.class);
-            if (step == null) { // compatibility: used to be transient
-                listener.getLogger().println("Queue item for node block in " + run.getFullDisplayName() + " is missing (perhaps JENKINS-34281), but cannot reschedule");
+            if (state == null) {
+                Run<?, ?> run = getContext().get(Run.class);
+                LOGGER.fine(() -> "No ExecutorStepDynamicContext found for node block in " + run + "; perhaps loading from a historical build record, hoping for the best");
```

Comment:
What is the expected behavior in this case? The various serialized pickles are rehydrated and the behavior is the same as it was before? Do we have any `@LocalData` tests that cover this case?

Reply:
The only difference in behavior should be for the cases formerly called out below with the JENKINS-34281 warning. In normal cases, this method would have `return`ed earlier. So I decided not to bother keeping the complicated code which should only apply after the one-time upgrade and even then only if something weird was broken.

Reply:
Ok, seems fine then. I think it would be nice though to have a happy-path `@LocalData` test to verify behavior across the upgrade, even if we just keep it for a few weeks and then delete it.

Reply:
https://github.com/jenkinsci/workflow-cps-plugin/blob/f82abbe3e8a5c749f05b2ad77d0707a0a6d0db0a/src/test/java/org/jenkinsci/plugins/workflow/cps/SerialFormTest.java#L76-L98 covers this code path.